test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.test2code.coverage
+
+import mu.*
+import java.util.*
+import java.util.concurrent.*
+import java.util.concurrent.atomic.*
+
+
+interface RetentionQueue {
+    fun addAll(data: Sequence<ByteArray>)
+    fun flush(): Sequence<ByteArray>
+}
+
+class InMemoryRetentionQueue(
+    private var queue: Queue<ByteArray> = ConcurrentLinkedQueue(),
+    private val totalSizeByteLimit: Long,
+) : RetentionQueue {
+    private val logger = KotlinLogging.logger("com.epam.drill.test2code.coverage.InMemoryRetentionQueue")
+    private val sizeOfQueue: AtomicLong = AtomicLong(0)
+
+    override fun addAll(data: Sequence<ByteArray>) {
```

Comment:
We need to discuss this function implementation. Currently I see a lot of issues:
1. Comparing size of a single `ByteArray` against total `totalSizeByteLimit` is wrong. We want to perform check against _actual sum of all elements size in bytes_, not against the limit value

2. `data.filter { it.size < totalSizeByteLimit }` will cause loss of any elements over `totalSizeByteLimit` (again - wrong)

3. `sizeOfQueue`  is set incorrectly - it must reflect total sum of sizes of `ByteArray`-s stored in the queue

Reply:
refactored